app/models/subject_queue.rb

```diff
@@ -84,18 +84,16 @@ def self.create_for_user(workflow, user, set_id: nil)
     end
   end
 
-  # NOTE: if the query scope returns a large set then
-  # don't use this query outside of a background Worker
-  # SQL optimisations welcome here...no postgres ops preseved order
   def self.dequeue_update(query, sms_ids)
-    query.find_each { |sq| sq.dequeue(sms_ids) }
+    return if sms_ids.blank?
+    dequeue_sql = "set_member_subject_ids = subarray(uniq(sort(set_member_subject_ids - array[?])), 0, #{DEFAULT_LENGTH})"
+    query.update_all([dequeue_sql, sms_ids])
   end
 
-  # NOTE: if the query scope returns a large set then
-  # don't use this query outside of a background Worker
-  # SQL optimisations welcome here...no postgres ops preseved order
   def self.enqueue_update(query, sms_ids)
-    query.find_each { |sq| sq.enqueue(sms_ids) }
+    return if sms_ids.blank?
+    enqueue_sql = "set_member_subject_ids = subarray(uniq(sort(set_member_subject_ids | array[?])), 0, #{DEFAULT_LENGTH})"
```

Comment:
That's not happening anymore. Since we're not iterating through all the queues. We're only removing the seen before.